src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -488,6 +488,18 @@ const vector<Real>& SpatialPooler::getBoostedOverlaps() const
   return boostedOverlaps_;
 }
 
+/** helper method that checks SP output is stable for given configuration */
+bool checkUnstableParams_(SpatialPooler &sp) {
+  vector<UInt> input = {1,1,0,0,1,0}; //FIXME auto to size of SP's input
+  if(sp.getInputDimensions().size()>1) return true; //FIXME skip nD as I cannot set auto input correctly
+  vector<UInt> out1(sp.getNumColumns(), 0); 
+  vector<UInt> out2(sp.getNumColumns(), 0); 
+  sp.compute(input.data(), true, out1.data());
+  sp.compute(input.data(), true, out2.data());
+  //TODO should we add SP.reset() and call it here? 
```

Comment:
TODO: now, the single run through the values is "stored" in SP's memory. Should we implement something as `SP::reset()` (that just calls initialize with the same params) to avoid that?